src/pudl/transform/gridpathratoolkit.py

```diff
@@ -0,0 +1,156 @@
+"""Transformations of the GridPath RA Toolkit renewable generation profiles.
+
+Wind and solar profiles are extracted separately, but concatenated into a single table
+in this module, as they have exactly the same structure. The aggregation tables are also
+concatenated together.
+"""
+
+import pandas as pd
+from dagster import AssetCheckResult, AssetIn, asset, asset_check
+
+import pudl
+
+
+def _transform_capacity_factors(
+    capacity_factors: pd.DataFrame, utc_offset: pd.Timedelta
+) -> pd.DataFrame:
+    """Basic transformations that can be applied to many profiles.
+
+    - Construct a datetime column and adjust it to be in UTC.
+    - Reshape the table from wide to tidy format.
+    - Name columns appropriately.
+    """
+    # Convert timestamp to UTC before stacking!
+    # TODO: Ensure that we're using the same hour starting vs. ending convention
+    # here as we use in the FERC-714 and EPA CEMS data. Should this be reflected in
+    # the column names here and elsewhere, whatever the convention is?
+    datetime_cols = ["year", "month", "day", "hour"]
+    capacity_factors["datetime_utc"] = (
+        pd.DatetimeIndex(pd.to_datetime(capacity_factors.loc[:, datetime_cols]))
+        - utc_offset
+    )
+    capacity_factors = (
+        capacity_factors.drop(columns=datetime_cols)
+        .set_index("datetime_utc")
+        .stack()
+        .reset_index()
+    )
+    capacity_factors.columns = ["datetime_utc", "aggregation_key", "capacity_factor"]
+    capacity_factors = capacity_factors.astype({"aggregation_key": "string"})
+    return capacity_factors
+
+
+@asset(io_manager_key="pudl_io_manager")
+def core_gridpathratoolkit__hourly_aggregated_extended_capacity_factors(
+    raw_gridpathratoolkit__aggregated_extended_solar_capacity: pd.DataFrame,
+    raw_gridpathratoolkit__aggregated_extended_wind_capacity: pd.DataFrame,
+) -> pd.DataFrame:
+    """Transform raw GridPath RA Toolkit renewable generation profiles.
+
+    Concatenates the solar and wind capacity factors into a single table and turns the
+    aggregation key into a categorical column to save space.
+    """
+    pacific_standard_time = pd.Timedelta("-8h")
+    return pd.concat(
+        [
+            _transform_capacity_factors(
+                capacity_factors=raw_df, utc_offset=pacific_standard_time
+            )
+            for raw_df in [
+                raw_gridpathratoolkit__aggregated_extended_solar_capacity,
+                raw_gridpathratoolkit__aggregated_extended_wind_capacity,
+            ]
+            if not raw_df.empty
```

Comment:
This check for an empty dataframe is required in case any of the technology types isn't being processed.